gatsby-node.js

```diff
@@ -281,6 +283,7 @@ exports.createPages = async ({ actions }) => {
                     slug,
                     snootyStitchId: SNOOTY_STITCH_ID,
                     __refDocMapping: pageNodes,
+                    _xrefDocMapping: pageNodes,
```

Comment:
@sophstad I noticed something else interesting about gatsby. When querying in production (via graphql) for a page's "__refDocMapping", graphql for whatever reason when it creates a schema sets it to "_xrefDocMapping". Meaning when we query graphql directly (which we do for this plugin in `gatsby-config`), we have to query `_xrefDocMapping`, since it's part of the generated schema, but that isn't what the field is really referring to. When I added this line to explicitly set it, I had success.

I also got a warning at build time that this assignment was a duplicate, but the previous value was null from graphql. Any thoughts on the best way to move forward with this?

Reply:
When running on development, the same issue occurs. Here is the schema on development in graphiql.

<img width="278" alt="Screen Shot 2020-03-16 at 5 21 02 PM" src="https://user-images.githubusercontent.com/9064401/76800699-96c7d000-67aa-11ea-91a9-15e4bc6c96ad.png">

Reply:
Isn't this what we're using the stitch functions for, rather than querying gql? Can we use the same function we use to get all the `learn` articles to populate this list as well?

Reply:
Yes, we do use the stitch functions to query this data. The `gatsby-plugin-feed` plugin, however, runs after the pages are built.

My understanding is we pull data from stitch to get page data, then for each page run the `createPage` method. After all of the pages are built, the inner graphql schema is updated to represent the page nodes as well, which is where these values are represented and generalized into an internal schema for graphql.

It looks like [here](https://www.gatsbyjs.org/docs/gatsby-lifecycle-apis/#bootstrap-sequence) we see the graphql schema being generated before pages are built, and then updated right after. The `gatsby-plugin-feed` config is run during the `onPostBuild` step at the very end.

Since at this point in the build cycle our schema should be complete, the plugin expects graphql to be the source of truth for pages and **requires** a graphql query to fetch page data.

Of course, we could try to do this without the plugin, but not sure that is worth the effort. I'm just confused why this field is being renamed in the schema but isn't a problem for pages.

Reply:
> requires a graphql query to fetch page data

This isn't technically true, but is the preferred way.

I just don't see us using gql anywhere else on the site and use stitch as our source of truth. In the strictest technical sense we should be adding our stitch functions as a source plugin which would make the source of truth and gql match more closely.

Reply:
okay! I can dig into this a bit more here